sql/core/src/main/scala/org/apache/spark/sql/jdbc/OracleDialect.scala

```diff
@@ -68,5 +69,13 @@ private case object OracleDialect extends JdbcDialect {
     case _ => None
   }
 
+  override def beforeFetch(connection: Connection, properties: Map[String, String]): Unit = {
+    // Set general date and timestamp format before query.
+    val stmt = connection.createStatement()
+    stmt.execute("alter session set NLS_DATE_FORMAT = 'YYYY-MM-DD'")
+    stmt.execute("alter session set NLS_TIMESTAMP_FORMAT = 'YYYY-MM-DD HH24:MI:SS.FF'")
```

Comment:
Will this impact the existing users?
> val sqlText = s"SELECT $columnList FROM ${options.table} $myWhereClause"

`${options.table}` could be a subquery containing the timestamp/date, right?